### PR TITLE
Hotfix PR #1733

### PR DESCRIPTION
**Pull request by sectore**
to backport #1733 back to `master`.

As in https://docs.google.com/spreadsheets/d/1B9OrkY543L-N1wcCAPV4fHW20N89N-tahXUCLusOQo8/edit#gid=0 described, just few commits (but not all) have to be back ported. 

Please check this PR carefully (I know everyone is doing this :)), because I'm not fully deep into every detail of #1733 and not sure if we do really need everything here...

### CODE REVIEW

File: node/src/Pos/Wallet/Web/State/Acidic.hs [hunk 1]
```diff
@@ -66,6 +66,7 @@ module Pos.Wallet.Web.State.Acidic
        , RemoveNextUpdate (..)
        , InsertIntoHistoryCache (..)
        , RemoveFromHistoryCache (..)
+       , UpdateHistoryCache2 (..)
```
Comment (adinapoli-iohk) [inline node/src/Pos/Wallet/Web/State/Acidic.hs]: This doesn't seem to exist in the original PR. Why was it necessary?

File: node/src/Pos/Wallet/Web/State/Storage.hs [hunk 2]
```diff
@@ -447,8 +448,7 @@ testReset = put def
 -- Legacy transaction, no longer used. For existing Db tx logs only. Now use
 -- 'removeHistoryCache', 'insertIntoHistoryCache' or 'removeFromHistoryCache'
 updateHistoryCache :: CId Wal -> [TxHistoryEntry] -> Update ()
-updateHistoryCache cWalId cTxs =
-    wsHistoryCache . at cWalId ?= txHistoryListToMap cTxs
```
Comment (adinapoli-iohk) [inline node/src/Pos/Wallet/Web/State/Storage.hs]: I don't think we want this, as both `master` and `cardano-sl-1.0-exchanges` have the same implementation for `updateHistoryCache`:

``` haskell
-- Legacy transaction, no longer used. For existing Db tx logs only. Now use
-- 'removeHistoryCache', 'insertIntoHistoryCache' or 'removeFromHistoryCache'
updateHistoryCache :: CId Wal -> [TxHistoryEntry] -> Update ()
updateHistoryCache cWalId cTxs =
    wsHistoryCache . at cWalId ?= txHistoryListToMap cTxs
```

File: node/src/Pos/Wallet/Web/State/Storage.hs [hunk 3]
```diff
@@ -458,6 +458,10 @@ removeFromHistoryCache :: CId Wal -> Map TxId () -> Update ()
 removeFromHistoryCache cWalId cTxs =
     wsHistoryCache . at cWalId . non' _Empty %= (`M.difference` cTxs)
 
+updateHistoryCache2 :: CId Wal -> Map TxId TxHistoryEntry -> Update ()
```
Comment (adinapoli-iohk) [inline node/src/Pos/Wallet/Web/State/Storage.hs]: The entire `...Cache2` thing can go, I'm decently confident.

**Comment by dcoutts** [comment]
Yes, the `updateHistoryCache2` was obsoleted by #1722, which has already been backported to develop and master: e9b4a2433b0602a6a2dc5523ad531b6a5238b143.

**Pull request closed without merging**